scripts/pre_commit_linter.py

```diff
@@ -468,6 +468,57 @@ def _pre_commit_linter(all_files):
     return summary_messages
 
 
+def _check_newline_character(all_files):
+    """This function is used to check new line characters.
+    """
+    print 'Starting to check new line at EOF'
+    print '----------------------------------------'
+    total_files_checked = 0
+    total_error_count = 0
+    summary_messages = []
+    all_files = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)]
+    failed = False
+    for filename in all_files:
+        with open(filename, 'rb+') as f:
+            total_files_checked += 1
+            chars = 0
+            for line in f:
+                chars += len(line)
```

Comment:
When reporting an error, please _always_ include an error traceback -- I don't have access to your environment, and it's very hard to figure out what is going on without this information.

I suggest fixing and pushing so that the errors show up on Travis; alternatively, take a careful look at the logs and see if you can identify what's going on.